index.html

```diff
@@ -526,14 +536,150 @@ <h3 id="installation-sec">
+          <h4>
+            <code>prompt()</code> method
+          </h4>
+          <p>
+            The <dfn>prompt</dfn> method, when called, runs the following
+            steps:
+          </p>
+          <ol>
+            <li>If <var>this</var>.<a>[[\userResponsePromise]]</a> is pending:
+              <ol>
+                <li>If this event's <a
+                  data-cite="!DOM#dom-event-istrusted"><code>isTrusted</code></a>
+                  attribute is <code> false</code>, reject
+                  <var>this</var>.<a>[[\userResponsePromise]]</a> with
+                  <a>NotAllowedError</a>, optionally informing the developer
+                  that untrusted events can't call <code>prompt()</code>.
+                </li>
+                <li>Else if <var>this</var>.<a>[[\didPrompt]]</a> is
+                  <code>false</code>, set <var>this</var>.<a>[[\didPrompt]]</a>
+                  to <code>true</code>, then <a>in parallel</a>, <a>request to
+                  present an install prompt</a> with this event. Wait, possibly
+                  indefinitely, for the end-user to make a choice.
+                </li>
+              </ol>
+            </li>
+            <li>Return <var>this</var>.<a>[[\userResponsePromise]]</a>.
+            </li>
+          </ol>
+          <p>
+            To <dfn>request to present an install prompt</dfn> with
+            <a>BeforeInstallPromptEvent</a> <var>event</var>:
+          </p>
+          <ol>
+            <li>
+              <a>Present an install prompt</a> and let <var>outcome</var> be
+              the result.
+            </li>
+            <li>Let <var>responseObj</var> be a newly created
+            <a>PromptResponseObject</a> whose <a
+            for="PromptResponseObject">userChoice</a> member is the value of
+            <var>outcome</var>.
+            </li>
+            <li>Resolve <var>event</var>.<a>[[\userResponsePromise]]</a> with
+            <var>responseObj</var>.
+            </li>
+          </ol>
+        </section>
+        <section class="informative">
+          <h4>
+            Usage example
+          </h4>
+          <p>
+            This example shows how one might prevent an automated install
+            prompt from showing until the user clicks a button to install the
+            app. In this way, the site can control the installation experience.
+          </p>
+          <pre class="example"
+               title="Using beforeinstallprompt to present an install button">
+            window.addEventListener("beforeinstallprompt", async (event) =&gt; {
+              // Suppress automatic prompting.
+              event.preventDefault();
+
+              // Show the (disabled-by-default) install button. This button
+              // resolves the installButtonClicked promise when clicked.
+              installButton.disabled = false;
+
+              // Wait for the user to click the button.
+              await installButtonClicked;
```

Comment:
Hmm yeah. async/await is nice but a bit magic here.

Refactored into a traditional addEventListener setup. The outer function no longer needs to be async.